src/components/publishing/display/canvas/text.tsx

```diff
@@ -0,0 +1,112 @@
+import { compact, map } from "lodash"
+import React from "react"
+import styled, { StyledFunction } from "styled-components"
+import { pMedia } from "../../../helpers"
+import Fonts from "../../fonts"
+
+interface DisplayCanvasTextProps extends React.HTMLProps<HTMLDivElement> {
+  disclaimer?: any
+  unit: any
+}
+
+interface DivProps extends React.HTMLProps<HTMLDivElement> {
+  layout?: any
+  isSlideshowWithCaption?: boolean
+}
+
+const DisplayCanvasText: React.SFC<DisplayCanvasTextProps> = props => {
+  const { disclaimer, unit } = props
+  const isSlideshowWithCaption = unit.layout === "slideshow" && compact(map(unit.assets, "caption")).length
+  const hasDisclaimer = unit.layout !== "overflow" && !isSlideshowWithCaption
+  return (
+    <CanvasInner layout={unit.layout} isSlideshowWithCaption>
+      <Logo layout={unit.layout} src={unit.logo} />
+      <div>
+        <Headline layout={unit.layout}>{unit.headline}</Headline>
+        <Link layout={unit.layout}>{unit.link.text}</Link>
+      </div>
+      {hasDisclaimer && disclaimer}
+    </CanvasInner>
+  )
+}
+
+const Div: StyledFunction<DivProps> = styled.div
+const Img: StyledFunction<DivProps> = styled.img
+
+const Logo = Img`
+  width: 100%;
+  height: 100%;
+  object-fit: contain;
+  max-width: ${props => (props.layout === "overlay" ? "300px;" : "185px;")}
+  max-height: ${props => (props.layout === "overlay" ? "100px;" : "65px;")}
+  margin: ${props => (props.layout === "overlay" ? "60px auto;" : "20px 0;")}
+  ${props => pMedia.md`
+    object-position: left;
+    width: auto;
+    ${props.layout === "overlay" &&
+      `max-width: calc(100% - 60px);
+        margin: 15px auto;`}
+  `}
+`
+const CanvasInner = Div`
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: ${props =>
+    props.layout === "overlay" || props.isSlideshowWithCaption ? "space-around;" : "space-between;"}
+  ${props =>
+    props.layout === "overlay" &&
+    `text-align: center;
+        color: #fff;
+        max-width: 500px;`}
+  ${props => pMedia.md`
+    ${props.layout === "slideshow" && "width: 100%;"}
+  `}
+  ${pMedia.sm`
+    max-width: 100%;
+    width: 100%;
+    padding: 0;
+  `}
+`
+const Headline = Div`
+  ${props => (props.layout === "overlay" ? Fonts.garamond("s23") : Fonts.garamond("s40"))}
+  margin-bottom: 25px;
+  line-height: ${props => (props.layout === "overlay" ? "1.35em;" : "1.1em;")} 
+  ${props => pMedia.lg`
+    ${props.layout !== "overlay" && Fonts.garamond("s23")}
+  `}
+  ${props => pMedia.sm`
+    ${props.layout === "overlay"
+      ? `margin: 0 auto 10px auto;
+         padding: 20px;`
+      : `margin-bottom: 10px;
+         max-width: calc(100% - 40px);`}
```

Comment:
This totally works, but just as a suggestion I've found that when styled-component prop logic gets wild, it helps to move logic into a function like, `getMobileHeadlineStyle(props)` which makes it a bit easier on the eye.